### PR TITLE
feat(settings): Adding new Backup Seed Phrase section & modal

### LINKED ISSUE #4918: New Backup Seed Phrase section & modal
### Description

See designs at https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/%E2%9A%99%EF%B8%8F-Settings-%7C-Desktop?node-id=884%3A94667

* [ ] remove old "Backup Seed Phrase" currently in the privacy and security section

* [ ] Add a new "Backup Seed Phrase" section to the profile sidebar. this should only be visible if the seed phrase has not need backed up yet. clicking on this section should open the backup seed phrase modal

Screen Shot 2022-02-26 at 1 28 05 PM

* [ ] Update the modal wizard to match the new designs

Screen Shot 2022-02-26 at 1 29 01 PM

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #4918

### What does the PR do
Adds new Backup Seed Phrase section & modal

### Affected areas
profile settings

### Screenshot of functionality
https://user-images.githubusercontent.com/31625338/171409300-c2b8ea63-27a0-48b0-998f-45fcfbd06a08.mov

### CODE REVIEW

**Comment by alexandraB99** [comment]
@John-44 could you please check the screen recording and let me know if validations in step 2&3 are ok? I couldn't find anything in figma.

**Review by stefandunca — APPROVED** [review]
I like the simplicity, component structure and granularity. Great job.

I found it hard to follow the scope of the step components until I had a clear look and run it myself. This can be improved by using names instead of indexes e.g. `BackupSeedStepOne.qml` -> "ConfirmSeedPhrasePanel", `BackupSeedStepTwo.qml` -> "ConfirmFirstRandomWordPanel.qml" ...

`BackupSeedStepOne.qml` and `BackupSeedStepTwo.qml` are duplicate code. What do you think about using a generic component and fill in the particular properties and process events? In case of changing something in the confirmed layout one has to mind the second step and adjust it. For other maintainers or far future you, it might require a second debugging round to be obvious.

File: ui/app/AppLayouts/Profile/popups/backupseed/BackupSeedStepTwo.qml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import StatusQ.Controls.Validators 0.1
+
+BackupSeedStep {
+    property int firstRandomNo
+    property string wordAtFirstRandomNo
+    property bool firstWordValid: true
+    function validateFirstWord() {
+        if (!!inputText.text)
+            return (wordAtFirstRandomNo === inputText.text);
+        return true;
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Profile/popups/backupseed/BackupSeedStepTwo.qml:13]: I know, looking at [figma](https://www.figma.com/file/idUoxN7OIW2Jpp3PMJ1Rl8/⚙%EF%B8%8F-Settings-%7C-Desktop?node-id=884%3A94667) "Continue" button is enabled even with no words added, I have asked John above for clarifications.

**Review by borismelnik — APPROVED** [review]

**Comment by John-44** [comment]
Have just reviewed the screencast and when the user gets to Step 2 at 0:07 in the video, the 'Continue' button should not be in the active state until the user has correctly entered the word.  The rest looks good 🙂

File: ui/app/AppLayouts/Profile/popups/BackupSeedModal.qml [hunk 2]
```diff
@@ -1,293 +1,122 @@
...
-                                anchors.left: count.right
-                                anchors.leftMargin: Style.current.padding
-                                selectByMouse: true
-                                readOnly: true
-                            }
-                        }
+    rightButtons: [
+        StatusFlatButton {
+            text: "Not Now"
+            visible: (contentLoader.item.objectName === "acknowledgment")
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/AppLayouts/Profile/popups/BackupSeedModal.qml:27]: This particular case will not make a big difference but it's the first thing I look into when using indexes and hardcoded strings. The approach with states and driving the step bar and panels based on the StateMachine looks like an improvement.
However, for the example's sake, I would go with a singleton component in `backupseed` folder:
`backupseed/Definitions.qml`
```qml
pragma Singleton

QtObject {
    property string acknowledgmentId: "ackn"
    ...
    
    // If enum are required
    enum StepIndex {
        IntroStep,
        FirstCheck,
        SecondCheck,
    }
}
```

**Comment by stefandunca** [comment]
> @stefandunca was also thinking about that but was a bit saturated, tbh. files are renamed and updated now, you can have a second look if you like!

@alexandraB99 no worry, great job.

**Comment by Hbouaz** [comment]
Tested:
+ John's comment above about activating buttons has been resolved
- New issue detected whereby the user can skip to last step and complete the process without revealing and validating the seed phrase 3/12 words, simply by clicking the 4th ad last step on the progress bar, see screen grab below:


https://user-images.githubusercontent.com/26457452/171433402-0fa02ec8-9b6a-4fab-98f5-4d4493a8db4a.mov

**Comment by alexandraB99** [comment]
> Tested:
> 
> * John's comment above about activating buttons has been resolved
> 
> * New issue detected whereby the user can skip to last step and complete the process without revealing and validating the seed phrase 3/12 words, simply by clicking the 4th ad last step on the progress bar, see screen grab below:
> 
>  seedPhraseModal.mov

@Hbouaz great finding, fixed!

**Review by Hbouaz — APPROVED** [review]
